zebra-chain/src/value_balance.rs

```diff
@@ -0,0 +1,297 @@
+//! A type that can hold the four types of Zcash value pools.
+
+use crate::amount::{Amount, Constraint, Error, NegativeAllowed, NonNegative};
+
+/// An amount spread between different Zcash pools.
+#[derive(Clone, Copy, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(bound = "C: Constraint")]
```

Comment:
Possibly unused code: Do we need `serde` here?

Zebra uses `ZcashSerialize` and `IntoDisk` for serialization, not `serde`.

```suggestion
#[derive(Clone, Copy, Debug, PartialEq)]
```